Source/ExecutionEngine/ConsolePrinter.cs

```diff
@@ -176,4 +178,25 @@ public virtual void ReportBplError(IToken tok, string message, bool error, TextW
       tw.WriteLine(s);
     }
   }
+
+  public void ReportImplementationsBeforeVerification(Implementation[] implementations) {
+    // Do not print anything to console
+  }
+
+  public void ReportStartVerifyImplementation(Implementation implementation) {
+    // Do not print anything to console
+  }
+
+  public void ReportEndVerifyImplementation(Implementation implementation, VerificationResult result) {
+    // Do not print anything to console
+  }
+
+  public int GetVerificationPriority(IToken implTok) {
+    return 0;
+  }
+
+  public void ReportAssertionBatchResult(Split split, Dictionary<AssertCmd, ConditionGeneration.Outcome> perAssertOutcome,
```

Comment:
I'm wondering whether `perAssertOutcome` should instead be a `VCResult`, as those are what make up the `VerificationResult` passed into `ReportEndVerifyImplementation`.

Also note that there's an `OnVCResult` method in `VerifierCallback`.

Reply:
Which of these do you have in mind?
- `Dictionary<AssertCmd, VCResult> perAssertOutcome`
- `VCResult perAssertOutcome`

`The first one would contain more information so it's definitely not a big deal, I can make it happily.
The second one would require to store more information in VCResult, which I can do, but I am wondering first if this is what you meant.`

Reply:
I'm thinking the latter, which includes all of the assertions in the batch. It only includes one outcome, though, so you'll still need the logic you have (somewhere) to use the counterexamples to figure out which of the assertions in a batch failed when the outcome indicates failure.

Reply:
I'll try to store the counter-examples in this VCResult so that it makes it easier.

Reply:
I ended up storing the counter-examples in the VCResult, and the code to figure out per-assertion outcome is now in a method of this VCResult that can be invoked.